src/components/Tables/Table.tsx

```diff
@@ -307,44 +308,51 @@ const Table = <Data,>({
         };
         columns.unshift({
             name: "",
-            cell: (row: Row<Data>) => (
-                <EditAndReorderArrowDiv>
-                    {editableConfig.onSwapRows && (
-                        <StyledIconButton
-                            onClick={() => swapRows(row.rowId, true)}
-                            aria-label="reorder row upwards"
-                            disabled={isSwapping}
-                        >
-                            <StyledIcon icon={faAnglesUp} />
-                        </StyledIconButton>
-                    )}
-                    {editableConfig.onEdit && (
-                        <StyledIconButton
-                            onClick={() => editableConfig.onEdit!(row.rowId)}
-                            aria-label="edit"
-                        >
-                            <StyledIcon icon={faPenToSquare} />
-                        </StyledIconButton>
-                    )}
-                    {editableConfig.onSwapRows && (
-                        <StyledIconButton
-                            onClick={() => swapRows(row.rowId, false)}
-                            aria-label="reorder row downwards"
-                            disabled={isSwapping}
-                        >
-                            <StyledIcon icon={faAnglesDown} />
-                        </StyledIconButton>
-                    )}
-                    {editableConfig.onDelete && (
-                        <StyledIconButton
-                            onClick={() => editableConfig.onDelete!(row.rowId)}
-                            aria-label="delete"
-                        >
-                            <StyledIcon icon={faTrashAlt} />
-                        </StyledIconButton>
-                    )}
-                </EditAndReorderArrowDiv>
-            ),
+            cell: (row: Row<Data>) => {
+                const isRowDeletable =
+                    editableConfig.onDelete &&
+                    ("id" in row.data && typeof row.data.id === "string"
+                        ? !editableConfig.undeletableIds?.includes(row.data.id)
+                        : true);
```

Comment:
Instead of checking the "id" field in the data internally, why don't we 
- define a prop `isDeletable: (row: Data) => boolean`
- use that prop to determine if the row is deletable
- let the parent define which row is deletable

This gives us the flexibility to define which one is deletable in the parent in any way we want, not just using the "id" field.

Reply:
that makes a lot of sense, thank you! have done